src/popups/ConfirmWindow/ConfirmWindow.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { confirmable } from 'react-confirm';
+import classNames from 'classnames';
+
+import styles from './ConfirmWindow.module.css';
+
+export const ConfirmWindow = confirmable(({ show, proceed, conformation }) => {
+    return (
+        <div
+            className={classNames(styles.windowContainer, {
+                [styles.notActive]: !show,
+            })}
+        >
+            <div className={styles.window}>
+                <div className={styles.text}>{conformation}</div>
+                <div className={styles.btnContainer}>
+                    <button
+                        className={styles.btn}
+                        onClick={() => proceed(false)}
```

Comment:
Here and below

This is an antipattern. What if we would like to do *more* when the button is clicked? Why would we need to subscribe to some function, when we may pass the callback from the very beginning?

My suggestion is:
1. Instead of `proceed` pass `onCancel` and `onConfirm` callbacks
2. replace `() => proceed(xxx)` with just `onXxx`
3. Handle the logic outside this component

Reply:
If it's a library syntax - just skip this thread (but it would still look ugly to me XD)